source/server/hot_restarting_parent.cc

```diff
@@ -82,6 +84,19 @@ void HotRestartingParent::onSocketEvent() {
 
 void HotRestartingParent::shutdown() { socket_event_.reset(); }
 
+HotRestartingParent::Internal::Internal(Server::Instance* server) : server_(server) {
+  // Track the hot-restart generation. Using gauge's accumulate semantics,
+  // the increments will be combined across hot-restart. This may be useful
+  // at some point, though the main motivation for this stat is to enable
+  // an integration test showing that dynamic stat-names can be coalesced
+  // across hot-restarts. There's no other reason this particular stat-name
+  // needs to be created dynamically.
+  Stats::StatNameDynamicPool pool(server_->stats().symbolTable());
+  Stats::Gauge& gauge = server_->stats().gaugeFromStatName(pool.add("hotrestart.generation"),
```

Comment:
Maybe put this under the `server.` tree and can you also document with the other server stats (I'm not sure where that is off the top of my head)?

Reply:
done; also renamed to match existing style. However I couldn't add it to the server-stats block via the macros as then they wouldn't be created dynamically, defeating the integration test coverage I was trying to add.